addons/web/static/tests/public/interaction_service.test.js

```diff
@@ -0,0 +1,335 @@
+import { describe, expect, test } from "@odoo/hoot";
+import { queryOne } from "@odoo/hoot-dom";
+import { animationFrame } from "@odoo/hoot-mock";
+
+import { Component, xml } from "@odoo/owl";
+import { makeMockEnv } from "@web/../tests/web_test_helpers";
+import { Interaction } from "@web/public/interaction";
+import { startInteraction } from "./helpers";
+
+describe.current.tags("interaction_dev");
+
+test("properly handles case where we have no match for wrapwrap", async () => {
+    const env = await makeMockEnv();
+    expect(env.services["public.interactions"]).toBe(null);
+});
+
+test("wait for translation before starting interactions", async () => {
+    class Test extends Interaction {
+        static selector = ".test";
+
+        setup() {
+            expect("localization" in this.services).toBe(true);
+        }
+    }
+    await startInteraction(Test, `<div class="test"></div>`);
+});
+
+test("starting interactions twice should only actually do it once", async () => {
+    let n = 0;
+    class Test extends Interaction {
+        static selector = ".test";
+
+        setup() {
+            n++;
+        }
+    }
+
+    const { core } = await startInteraction(Test, `<div class="test"></div>`);
+
+    expect(n).toBe(1);
+    core.startInteractions();
+    await animationFrame();
+    expect(n).toBe(1);
+});
+
+test("start interactions even if there is a crash", async () => {
+    class Boom extends Interaction {
+        static selector = ".test";
+
+        setup() {
+            expect.step("start boom");
+            throw new Error("boom");
+        }
+        destroy() {
+            expect.step("destroy boom");
+        }
+    }
+    class NotBoom extends Interaction {
+        static selector = ".test";
+
+        setup() {
+            expect.step("start notboom");
+        }
+        destroy() {
+            expect.step("destroy notboom");
+        }
+    }
+
+    const { core } = await startInteraction([Boom, NotBoom], `<div class="test"></div>`, {
+        waitForStart: false,
+    });
+
+    let e = null;
+    try {
+        await core.isReady;
+    } catch (_e) {
+        e = _e;
+    }
+    expect(e.message).toBe("boom");
```

Comment:
This should do the trick in a nicer manner. Check other occurrences for similar patterns
```suggestion
    await expect(core.isReady).rejects.toThrow("boom");
```